{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SC-VAR Usage\n",
    "\n",
    "single cell ATAC-seq data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We assign noncoding SNPs to their target genes based on chromatin accessibility in disease-relevant tissues measured by single-cell ATAC sequence. \n",
    "Which can be used to find disease risk genes and pathways.\n",
    "\n",
    "And by calculating the single-cell chromatin accessibility disease score \n",
    "\n",
    "Users can infer cell types involved in complex traits and diseases using single-cell epigenomes.\n",
    "\n",
    "[ This branch does not rely on fine cell annotations and other Omics data. ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!pip install sc-var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "def generate_atac_risk_file(overlap_matrix, scemagma_result, output_file=None):\n",
    "    \"\"\"\n",
    "    Generate the risk file (CRE_ID, ZSTAT) required for scRiskDB ATAC-seq analysis.\n",
    "    \n",
    "    Parameters\n",
    "    ----------\n",
    "    overlap_matrix : pandas.DataFrame\n",
    "        Output from scv.snp_peak() function. \n",
    "        Contains the mapping between SNPs and Peaks.\n",
    "        \n",
    "    scemagma_result : str\n",
    "        Path to the output file from scemagma pipeline gene analysis (the .genes.out file).\n",
    "        \n",
    "    output_file : str, optional\n",
    "        Path to save the resulting CSV file (e.g., 'risk_data.csv').\n",
    "        \n",
    "    Returns\n",
    "    -------\n",
    "    pandas.DataFrame\n",
    "        A DataFrame with columns ['CRE_ID', 'ZSTAT']\n",
    "    \"\"\"\n",
    "    \n",
    "\n",
    "    cmagma_gene = pd.read_csv(scemagma_result, sep='\\s+') \n",
    "    \n",
    "    if 'GENE' in cmagma_gene.columns:\n",
    "        cmagma_gene = cmagma_gene.sort_values('P').drop_duplicates(subset=['GENE'], keep='first')\n",
    "    \n",
    "    cmagma_gene_sig = cmagma_gene[cmagma_gene['P'] < 0.05]\n",
    "\n",
    "    filtered_matrix = overlap_matrix[overlap_matrix['gene'].isin(cmagma_gene_sig['GENE'])] # æˆ– 'symbol'\n",
    "    \n",
    "    if filtered_matrix.empty:\n",
    "        print(\"Warning: No peaks linked to significant risk genes found.\")\n",
    "        return pd.DataFrame(columns=['CRE_ID', 'ZSTAT'])\n",
    "\n",
    "    peak_weights = get_peak_weights(filtered_matrix)\n",
    "\n",
    "    risk_df = peak_weights.copy()\n",
    "\n",
    "    risk_df = risk_df.rename(columns={\n",
    "        risk_df.columns[0]: 'CRE_ID', \n",
    "        'weight': 'ZSTAT'\n",
    "    })\n",
    "    \n",
    "\n",
    "    risk_df = risk_df[['CRE_ID', 'ZSTAT']]\n",
    "    \n",
    "\n",
    "    if output_file:\n",
    "        risk_df.to_csv(output_file, index=False)\n",
    "        print(f\"Risk file saved to: {output_file}\")\n",
    "        \n",
    "    return risk_df\n",
    "\n",
    "def get_peak_weights(overlap_matrix):\n",
    "\n",
    "    '''\n",
    "    For each peak, determine the lowest p-value of all SNPs that overlap with it.\n",
    "    '''\n",
    "\n",
    "    peak_weights = overlap_matrix.groupby(['chr', 'start', 'end']).agg({'pval': 'min'})\n",
    "    #if min p is 0, replace it with 1e-10\n",
    "    peak_weights['pval']=peak_weights['pval'].replace(0,1e-10)\n",
    "    overlap_matrix['zscore']= np.abs(scipy.stats.norm.ppf(overlap_matrix['pval']/ 2))\n",
    "    #keep zcore corresponding to min p\n",
    "    peak_weights=pd.merge(peak_weights,overlap_matrix[['chr','start','end','zscore']],on=['chr','start','end'],how='left')\n",
    "    peak_weights.reset_index(inplace=True)\n",
    "    peak_weights.rename({'zscore': 'weight'}, axis=1, inplace=True)\n",
    "    peak_weights['GENE']=peak_weights['chr'].astype(str)+'-'+peak_weights['start'].astype(str)+'-'+peak_weights['end'].astype(str)\n",
    "\n",
    "    peak_weights.sort_values(['weight'],ascending=False,inplace=True)\n",
    "    peak_weights.drop_duplicates(subset=['GENE'],keep='first',inplace=True)\n",
    "    peak_weights=peak_weights[['GENE','weight']]\n",
    "    return peak_weights"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import scdrs\n",
    "import scanpy as sc\n",
    "from anndata import AnnData\n",
    "from anndata import read_h5ad\n",
    "from scipy import stats\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sc_var\n",
    "from sc_var import method as scv\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib import cm\n",
    "from matplotlib import axes\n",
    "import pylab\n",
    "from matplotlib.pyplot import rc_context\n",
    "import os\n",
    "import warnings\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib as mpl\n",
    "import episcanpy as epi\n",
    "from cycler import cycler\n",
    "\n",
    "warnings.filterwarnings(\"ignore\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## preprocessing data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "GWAS SNP data format:\n",
    "\n",
    "chr pos rsids pval\n",
    "\n",
    "chr1    14773   rs878915777 0.13\n",
    "\n",
    "\n",
    "\n",
    "==============\n",
    "\n",
    "Peak data format:\n",
    "\n",
    "gene chr start end\n",
    "\n",
    "TTLL10  chr1    826700\t827679"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#GWAS data\n",
    "snp_list=scv.read_gwas('~/data/SCZ.txt')\n",
    "\n",
    "#IF only have the cicero_conn and cds fdata\n",
    "#Use the function help to get the peak-gene conn from cicero_conn and cds fdata\n",
    "peak_list=scv.get_p2g_conn('~/data/conn_path','~/data/fdata_path')\n",
    "#Or             \n",
    "#peak data ( Co-accessibility networks )\n",
    "peak_list=scv.load_peak_data('~/data/atac_cis_coacess.txt')\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Find the snp-peak overlap (Take times if data is large)\n",
    "\n",
    "!! ATTENTION [choose the right reference genome]\n",
    "\n",
    "check gencode webset for annotation version ( for now v43 for hg19 v44 for hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#overlp SNPs and peaks \n",
    "overlap_matrix = scv.snp_peak(peak_list,snp_list)\n",
    "#find snp inside the gene body (load the annotatio file from the original magma pipline)\n",
    "magma = scv.dmagma('~/data/magma0.genes.annot')\n",
    "#get gene coordinates\n",
    "corr = scv.gene_corr('~/data/gencode.v44.annotation.gff3.gz',peak_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get peak-snp-gene annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#SCZ as example\n",
    "anno=scv.annotate(overlap_matrix,corr,magma,'Schizophrenia.')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Gene Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Gene analysis\n",
    "#choose the 1000 genome reference panel map your GWAS data\n",
    "!magma\\\n",
    "    --bfile /g1000_eur/g1000_eur \\                     \n",
    "    --pval scz.txt use='rsids,pval' N=328748 \\\n",
    "    --gene-annot /Schizophrenia.scemagma.genes.annot \\                                                   \n",
    "    --out /output/scemagma\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# You need overlap_matrix\n",
    "# And From MAGMA,you Got '<Disease>.genes.out'\n",
    "\n",
    "scriskdb_riskfile=generate_atac_risk_file(\n",
    "    overlap_matrix = overlap_matrix,\n",
    "    scemagma_result = 'scemagma.genes.out',\n",
    "    output_file = 'risk_for_upload.csv'\n",
    ")\n",
    "\n",
    "# Output: SCZ_risk_for_upload.csv created.\n",
    "# Now upload this file to scRiskDB!"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bio",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
